src/Evolution/Systems/Ccz4/TimeDerivative.cpp

```diff
@@ -0,0 +1,605 @@
+      lapse_times_d_a_tilde, (*lapse)() * d_a_tilde(ti_k, ti_i, ti_j));
+
+  ::TensorExpressions::evaluate<ti_k>(lapse_times_field_a,
+                                      (*lapse)() * field_a(ti_k));
+
+  ::TensorExpressions::evaluate<ti_i, ti_j>(
+      lapse_times_conformal_spatial_metric,
+      (*lapse)() * conformal_spatial_metric(ti_i, ti_j));
+
+  ::TensorExpressions::evaluate(
+      lapse_times_ricci_scalar_plus_divergence_z4_constraint,
+      (*lapse)() * (*ricci_scalar_plus_divergence_z4_constraint)());
+
+  ::TensorExpressions::evaluate<ti_I>(shift_times_deriv_gamma_hat,
+                                      shift(ti_K) * d_gamma_hat(ti_k, ti_I));
+
+  ::TensorExpressions::evaluate<ti_i, ti_j>(
+      inv_tau_times_conformal_metric,
+      one_over_relaxation_time * conformal_spatial_metric(ti_i, ti_j));
+
+  // time derivative computation: eq 12a - 12m
+
+  // eq 12a : time derivative of the conformal spatial metric
+  ::TensorExpressions::evaluate<ti_i, ti_j>(
+      dt_conformal_spatial_metric,
+      2.0 * shift(ti_K) * field_d(ti_k, ti_i, ti_j) +
+          (*conformal_metric_times_field_b)(ti_i, ti_j) +
+          (*conformal_metric_times_field_b)(ti_j, ti_i) -
+          2.0 * one_third * conformal_spatial_metric(ti_i, ti_j) *
+              (*contracted_field_b)() -
+          2.0 * (*lapse)() *
+              (*a_tilde_minus_one_third_conformal_metric_times_trace_a_tilde)(
+                  ti_i, ti_j) -
+          (*inv_tau_times_conformal_metric)(ti_i, ti_j) *
+              ((*det_conformal_spatial_metric)() - 1.0));
+
+  // eq 12b : time derivative of the natural log of the lapse
+  ::TensorExpressions::evaluate(
+      dt_ln_lapse,
+      shift(ti_K) * field_a(ti_k) -
+          (*lapse_times_slicing_condition)() * (*k_minus_k0_minus_2_theta_c)());
+
+  // eq 12c : time derivative of the shift
+  // if s == 0
+  if (not evolve_shift) {
+    for (auto& component : *dt_shift) {
+      component = 0.0;
+    }
+  } else {
+    ::TensorExpressions::evaluate<ti_I>(
+        dt_shift, f * b(ti_I) + shift(ti_K) * field_b(ti_k, ti_I));
+  }
+
+  // eq 12d : time derivative of the natural log of the conformal factor
+  ::TensorExpressions::evaluate(
+      dt_ln_conformal_factor,
+      shift(ti_K) * field_p(ti_k) +
+          one_third * ((*lapse)() * trace_extrinsic_curvature() -
+                       (*contracted_field_b)()));
+
+  // eq 12e : time derivative of the trace-free part of the extrinsic curvature
+  ::TensorExpressions::evaluate<ti_i, ti_j>(
+      dt_a_tilde,
+      shift(ti_K) * d_a_tilde(ti_k, ti_i, ti_j) +
+          (*conformal_factor_squared)() *
+              ((*lapse)() * ((*spatial_ricci_tensor)(ti_i, ti_j) +
+                             (*grad_spatial_z4_constraint)(ti_i, ti_j) +
+                             (*grad_spatial_z4_constraint)(ti_j, ti_i)) -
+               (*grad_grad_lapse)(ti_i, ti_j)) -
+          one_third * conformal_spatial_metric(ti_i, ti_j) *
```

Comment:
In eq 12e in the paper, it looks to me like it's both multiplied and divided by the square of the conformal factor, and since it canceled, I neither multiplied or divided by it and just omitted it: ~phi^2~ * (1/3)  * (conformal_metric / ~phi^2~) * [...]     --->     (1/3)  * conformal_metric * [...]

Let me know if I'm missing something

Reply:
In eq 12e in the paper, it looks to me like it's both multiplied and divided by the square of the conformal factor, and since it cancels, I neither multiplied nor divided by it and just omitted it: ~phi^2~ * (1/3)  * (conformal_metric / ~phi^2~) * [...]     --->     (1/3)  * conformal_metric * [...]

Let me know if I'm missing something

Reply:
:( yea... Sorry!